gazebo/common/HeightmapData.cc

```diff
@@ -95,6 +112,41 @@ HeightmapData *HeightmapDataLoader::LoadTerrainFile(
     return LoadDEMAsTerrain(_filename);
   }
 }
+
+//////////////////////////////////////////////////
+/// Load terrain files taking into account the spherical
+/// coordinates surface type.
+HeightmapData *HeightmapDataLoader::LoadTerrainFile(
+    const std::string &_filename,
+    common::SphericalCoordinatesPtr _sphericalCoordinates)
```

Comment:
Hmm, wouldn't that be the section that runs when we don't have ``GDAL`` ? Since this overload is meant for the original ``LoadTerrainFile(const std::string &_filename)``, shouldn't it live in the same ``if`` block ?

Reply:
the header file defines [both overloads](https://github.com/osrf/gazebo/blob/fd3b266bf7e0ae96ba4597003c747a22c8a9df48/gazebo/common/HeightmapData.hh#L77-L94) regardless of whether GDAL is installed, so both overloads should be defined in the `#ifdef HAVE_GDAL` and `#else` blocks

this probably applies to `LoadDEMAsTerrain` as well, now that I think of it

I see the following related test failure on homebrew: https://build.osrfoundation.org/job/gazebo-ci-pr_any-homebrew-amd64/2817/testReport/(root)/INTEGRATION_heightmap/test_ran/

~~~
555: [ RUN      ] HeightmapTest.DartCollisionDetectorSelectionBullet
555: dyld: lazy symbol binding failed: Symbol not found: __ZN6gazebo6common19HeightmapDataLoader15LoadTerrainFileERKNSt3__112basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEEN5boost10shared_ptrINS0_20SphericalCoordinatesEEE
555:   Referenced from: /usr/local/lib/libgazebo_physics.11.dylib
555:   Expected in: flat namespace
555: 
555: dyld: Symbol not found: __ZN6gazebo6common19HeightmapDataLoader15LoadTerrainFileERKNSt3__112basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEEN5boost10shared_ptrINS0_20SphericalCoordinatesEEE
555:   Referenced from: /usr/local/lib/libgazebo_physics.11.dylib
555:   Expected in: flat namespace
555: 
555/676 Test #555: INTEGRATION_heightmap .................................Subprocess aborted***Exception:   1.38 sec
~~~

Reply:
Added overloads, but for ``LoadDEMAsTerrain``, I simply added a warning, as we can't use GDAL anyway in this case : https://github.com/osrf/gazebo/pull/3258/commits/38e0689a551182299fab778f5d7e0bbe0f14ce92